extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java

```diff
@@ -170,6 +192,77 @@ void dropNames(
         dropStaticResources.produce(new DropStaticResourcesBuildItem(resources));
     }
 
+    private Set<String> generateTracelessUris(final List<AnnotationInstance> annotations, final String rootPath) {
+        final Set<String> applicationUris = new HashSet<>();
+        for (AnnotationInstance annotation : annotations) {
```

Comment:
Documentation says:

> There are two ways to disable tracing for a specific REST endpoint.

I do not believe you can  disable tracing for specific REST endpoints based on paths, because you cannot identify REST resource methods with path. Path to resource method is `one-to-many` relation, but docs ATM says absolutely nothing about that.

I am worried about this annotation approach:

- What will happen if I annotate `@GET` method with certain path? Won't this will also exclude `@PUT` and any other HTTP method? Same must be true for different consumed/produced types etc. if I have 10 different endpoints with same path and I put this one one endpoint, is it expected that all the endpoints are not traced?
- If I placed this one `@Path("/hello/{id}") String method()` and I have another method `@Path("/hello/more-specific") completelyDifferentMethod()` then both endpoints will be excluded even though I only placed this on one method with less specific path?
- I can have endpoints only annotated with HTTP method like `@GET` etc. but this feature depends on `@Path` annotation, so here I cannot disable endpoint `get()`:
```
@Path("something")
public class SomethingResource {
  @GET
  public String get() {
   return "something";
 }

  @GET
  @Path("another")
  public String another() {
     return "another";
  }
}
```
- I think this won't work for subresources either, you will get illegal state for valid scenario like:
```
@Path("/simple")
public class SimpleQuarkusRestResource {
    @Traceless
    @Path("sub")
    public SubResource subResource() {
        return new SubResource();
    }

}

public class SubResource {

    @Path("123")
    @GET
    public String sub() {
        return "sub";
    }

    @GET
    @Path("otherSub")
    public String otherPath() {
        return "otherSub";
    }
}
```
- it does not consider inheritance, https://jakarta.ee/specifications/restful-ws/4.0/jakarta-restful-ws-spec-4.0#annotationinheritance, if I have `@Path` on parent class endpoint or interface method without `@Path`, I'll get exception message containing `Please ensure that the class is properly annotated with @Path annotation.`.
- documentation speaks about endpoints, not paths. REST clients has also `@Path`, what happens if I put this on the REST client? Or better, if I put this on REST endpoints, this will not exclude traces for the client as well? 

I suggest that someone who makes living working on Jakarta REST impl. like @geoand reviews this PR. Maybe it's alright, I wouldn't know.

Reply:
I believe that `@Traceless` should be removed completely as it's behavior is wrong. Allowing users to configure the path to not be traced makes sense, but with the way things are now, `@Traceless` is very misleading and get users into a lot of trouble.

Reply:
I want to make sure to properly acknowledge @mcruzdev for the work done here and @michalvavrik for raising this issue!

It's amazing to have such a great community that works so effectively on continuously improving Quarkus in so many ways!

Reply:
I want to make sure to properly acknowledge @mcruzdev for the work done here and @michalvavrik for raising this issue!

It's amazing to have such a great community that works so effectively on continuously improving Quarkus in so many ways!

Reply:
Thank you for catching this @michalvavrik. I really appreciate your reviews; I’ve been learning a lot.

We can remove the `@Traceless` annotation and rely solely on the configuration for now until we have a better implementation through annotations.

@geoand can I send a pull request for removing it? From what I’ve seen, this hasn’t been released yet, correct?

Reply:
cc: @brunobat

Reply:
Yes please do, this indeed has not been released yet.

Reply:
Was on a conference yesterday. I've replied in the new issue.